automated/health-checks/health-check_Core_SE.yaml

```diff
@@ -0,0 +1,43 @@
+device_type: Core_SE
+job_name: Core_SE - Standard health check
+priority: medium
+visibility: public
+
+timeouts:
+  job:
+    minutes: 5
+  action:
+    minutes: 1
+  connection:
+    minutes: 2
+
+actions:
+- deploy:
+    timeout:
+      # this is a health check, so no flashing or other time-intensive tasks
+      # need to be executed
+      minutes: 1
+    to: ssh
+
+- boot:
+    method: ssh
+    connection: ssh
+    # it is assumed that the revpi is in a working state for the health check,
+    # so the prompt should be set correctly (i.e. the hostname is correct)
+    prompts: ["root@RevPi[0-9]+:"]
```

Comment:
Applies to all device types, but will only comment once: does this need to be run as root / does it make sense to hard code the user in the prompt?

Reply:
According to @RamiGspo logging in as `pi` wasn't working. Maybe he has more info on that.

Reply:
Thank you Thomas, that's right... 

Until now this has not been possible, but maybe now I understand why this is so.
Probably this can be modified.
Now, using "pi@RevPi" we have this output:

```shell
Using device data host_address for primary connection.
scp-deploy Connecting to device root@RevPi32346.testing.revpi.kunbus.com using 'ssh -o Compression=yes -o PasswordAuthentication=no -o LogLevel=FATAL -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i /sshkey/lava_worker_ed25519 -p 22'

Setting prompt string to ['pi@RevPi[0-9]+:']
scp-deploy: Wait for prompt ['pi@RevPi[0-9]+:'] (timeout 00:01:00)
Linux RevPi32346 5.10.152-rt75-v8 #1 SMP PREEMPT_RT Fri, 16 Jun 2023 09:56:57 +0200 aarch64
The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.
Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Fri Jun 30 09:47:40 2023 from 10.42.11.150
scp-deploy timed out after 60 seconds
``` 

RevPi's device dictionary has this definition:

```shell
...
{% set ssh_host = "RevPi32346.testing.revpi.kunbus.com" %}
{% set ssh_user = "root" %}
{% set ssh_identity_file = '/sshkey/lava_worker_ed25519' %}

{% set device_prompt = "RevPi32346" %}
...
``` 

I will try some modifications and if it doesn't work I will write Tickets for this.

Reply:
The reason using user `pi` doesn't work is because the user isn't allowed to write files to `/`.

For LAVA to run a test, it needs to copy some files over to the DUT. These files are compressed in a `.tar.gz` file which is copied to `/`: https://lava.revpi.kunbus.com/scheduler/job/1652#L102
```
Calling: 'scp' '-o' 'Compression=yes' '-o' 'PasswordAuthentication=no' '-o' 'LogLevel=FATAL' '-P' '22' '-i' '/sshkey/id_ed25519' '-o' 'UserKnownHostsFile=/dev/null' '-o' 'StrictHostKeyChecking=no' '/var/lib/lava/dispatcher/tmp/1652/compress-overlay-cljv03wm/overlay-1.1.5.tar.gz' 'pi@192.168.168.140:/lava-1652-overlay-1.1.5.tar.gz'
```
After removing all the noise this just says `scp overlay.tar.gz pi@192.168.168.140:/overlay.tar.gz`. This, of course, fails due to missing permissions:
```
>> scp: /lava-1652-overlay-1.1.5.tar.gz: Permission denied
```

These files are then extracted to `/overlay`: https://lava.revpi.kunbus.com/scheduler/job/1615#L135
```
root@RevPi96995:~# tar  -C / -xzf /overlay.tar.gz
```

Why they are doing this in root of all places I don't know. Until we find a way around this copying of files to root, automatically giving the `pi` user the necessary permissions, changing the commands, or just execute parts of the job as user `root` we will have to keep using `root` as the user.